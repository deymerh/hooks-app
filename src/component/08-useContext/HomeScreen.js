import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const HomeScreen = () => {
  const { user, setUser } = useContext(UserContext);
  return (
    <div>
      <h1>HomeScreen</h1>
      <hr />
      <pre>{JSON.stringify(user, null, 2)}</pre>
      <div className="container-fluid">
        <button
          onClick={() => setUser({})}
          className="btn btn-outline-danger"
        >
          Logout
        </button>
      </div>
    </div>
  );
};
