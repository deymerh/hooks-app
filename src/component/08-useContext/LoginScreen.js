import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {
  const { setUser } = useContext(UserContext);
  const Login = () => {
    setUser({
      uid: 123456,
      name: 'Deymer',
      email: 'deymerh@hotmail.com'
    })
  }
  return (
    <div>
      <h1>LoginScreen</h1>
      <hr />
      <div className="btn btn-primary" onClick={Login}>Login</div>
    </div>
  );
};
