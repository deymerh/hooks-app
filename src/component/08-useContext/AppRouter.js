import React from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect
} from 'react-router-dom';
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import { Nabvar } from './Nabvar';

export const AppRouter = () => {
  return (
    <Router>
      <Nabvar />
      <div className="container mt-5">
        <Switch>
          <Route exact path="/" component={HomeScreen} />
          <Route exact path="/about" component={AboutScreen} />
          <Route exact path="/login" component={LoginScreen} />
          <Redirect to="/" />
        </Switch>
      </div>
    </Router>
  )
}
