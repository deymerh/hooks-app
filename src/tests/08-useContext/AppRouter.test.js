import { mount } from 'enzyme';
import { AppRouter } from '../../component/08-useContext/AppRouter';
import { UserContext } from '../../component/08-useContext/UserContext';

describe('Pruebas en el <AppRouter />', () => {
  const user = {
    id: 123,
    name: 'Deymer'
  }
  const wrapper = mount(
    <UserContext.Provider value={{
      user
    }}>
      <AppRouter />
    </UserContext.Provider>
  )

  test('Renderizarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
