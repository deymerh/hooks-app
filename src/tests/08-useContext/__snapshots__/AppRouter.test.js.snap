// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <AppRouter /> Renderizarse correctamente 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-primary"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        UseContext
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNav"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              aria-current="page"
              className="nav-link active"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              Home
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/about"
              onClick={[Function]}
            >
              About
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/login"
              onClick={[Function]}
            >
              Login
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>,
  <div
    className="container mt-5"
  >
    <div>
      <h1>
        HomeScreen
      </h1>
      <hr />
      <pre>
        {
  "id": 123,
  "name": "Deymer"
}
      </pre>
      <div
        className="container-fluid"
      >
        <button
          className="btn btn-outline-danger"
          onClick={[Function]}
        >
          Logout
        </button>
      </div>
    </div>
  </div>,
]
`;
